server.go

```diff
@@ -4099,11 +4213,51 @@ func (s *server) OutboundPeerConnected(connReq *connmgr.ConnReq, conn net.Conn)
 		s.removePeer(connectedPeer)
 		s.ignorePeerTermination[connectedPeer] = struct{}{}
 		s.scheduledPeerConnection[pubStr] = func() {
-			s.peerConnected(conn, connReq, false)
+			s.peerConnected(conn, connReq, false, access)
 		}
 	}
 }
 
+// assignPeerPerms assigns a new peer its permissions.
+func (s *server) assignPeerPerms(remotePub *btcec.PublicKey) (peerAccessStatus,
+	error) {
+
+	var access peerAccessStatus
+
+	shouldDc, dcErr := s.authGossiper.ShouldDisconnect(remotePub)
+	if dcErr != nil {
+		return access, dcErr
+	}
+
+	if shouldDc {
+		return access, ErrGossiperBan
+	}
+
+	remoteHex := hex.EncodeToString(remotePub.SerializeCompressed())
+
+	// Lock banScoreMtx for reading so that we can update the banning maps
+	// below.
+	s.banScoreMtx.RLock()
+	defer s.banScoreMtx.RUnlock()
+
+	if _, found := s.channelPeers[remoteHex]; found {
```

Comment:
I think should be in reverse, so pending check first, then channel peer check.

Reply:
no longer necessary